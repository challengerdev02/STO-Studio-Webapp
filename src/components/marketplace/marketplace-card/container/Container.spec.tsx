import { cleanup, queryByText, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import {
  CustomCardColoredStyledBox,
  CustomStyledBox,
  MarketplaceCardContainer,
  SpacedFlexContainer,
} from './index';

describe('Container', function () {
  afterEach(cleanup);
  let props = {
    width: '256px',
    height: '256px',

    hoverable: true,
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(
      <CustomStyledBox {...props} {...extraProps}>
        A content
      </CustomStyledBox>
    );
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should have Content "This is a test for Content"', () => {
    // @ts-ignore
    const { container } = setup();
    expect(queryByText(document.body, 'A content')).toBeInTheDocument();
  });

  it('should match with snapshot: SpacedFlexContainer', () => {
    // @ts-ignore
    const { container } = render(<SpacedFlexContainer />);
    expect(container).toMatchSnapshot();
  });

  it('should match with snapshot: CustomCardColoredStyledBox', () => {
    // @ts-ignore
    const { container } = render(<CustomCardColoredStyledBox position={20} />);
    expect(container).toMatchSnapshot();
  });

  it('should match with snapshot: CustomCardColoredStyledBox No Position', () => {
    // @ts-ignore
    const { container } = render(<CustomCardColoredStyledBox />);
    expect(container).toMatchSnapshot();
  });

  it('should match with snapshot: MarketplaceCardContainer', () => {
    // @ts-ignore
    const { container } = render(
      <MarketplaceCardContainer
        {...{
          coverImg:
            'https://robohash.org/a8cd9c187bb33c8a8f6e2774d71308fb?set=set4&bgset=&size=400x400',
          countDown: '01:52:09 left ðŸ”¥',
          startDate: new Date(2022, 0, 1, 12, 30, 20, 0).toDateString(),
          menuItems: ['Option A', 'Option B', 'Option c'],
          endDate: new Date(2021, 11, 16, 12, 30, 7, 999).toDateString(),
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should match with snapshot: MarketplaceCardContainer With Cover', () => {
    // @ts-ignore
    const { container } = render(
      <MarketplaceCardContainer
        {...{
          coverImg:
            'https://robohash.org/a8cd9c187bb33c8a8f6e2774d71308fb?set=set4&bgset=&size=400x400',
          countDown: '01:52:09 left ðŸ”¥',
          startDate: new Date(2022, 0, 1, 12, 30, 20, 0).toDateString(),
          menuItems: ['Option A', 'Option B', 'Option c'],
          cover: <div>Hello</div>,
          endDate: new Date(2021, 11, 16, 12, 30, 7, 999).toDateString(),
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should match with snapshot: MarketplaceCardContainer Without menuItems', () => {
    // @ts-ignore
    const { container } = render(
      <MarketplaceCardContainer
        {...{
          coverImg:
            'https://robohash.org/a8cd9c187bb33c8a8f6e2774d71308fb?set=set4&bgset=&size=400x400',
          countDown: '01:52:09 left ðŸ”¥',
          startDate: new Date(2022, 0, 1, 12, 30, 20, 0).toDateString(),
          cover: <div>Hello</div>,
          endDate: new Date(2021, 11, 16, 12, 30, 7, 999).toDateString(),
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
