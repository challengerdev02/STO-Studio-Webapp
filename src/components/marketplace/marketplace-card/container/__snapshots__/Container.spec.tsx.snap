// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match with snapshot 1`] = `
<div>
  <div
    class="sc-dkPtRN ktuioq"
    height="256px"
    width="256px"
  >
    A content
  </div>
</div>
`;

exports[`Container should match with snapshot: CustomCardColoredStyledBox 1`] = `
<div>
  <div
    class="sc-dkPtRN sc-hKwDye ekOsNC hkabfB"
  />
</div>
`;

exports[`Container should match with snapshot: CustomCardColoredStyledBox No Position 1`] = `
<div>
  <div
    class="sc-dkPtRN sc-hKwDye ekOsNC rBElr"
  />
</div>
`;

exports[`Container should match with snapshot: MarketplaceCardContainer 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered sc-eCImPb eXMxhc"
    countDown="01:52:09 left ðŸ”¥"
    coverImg="https://robohash.org/a8cd9c187bb33c8a8f6e2774d71308fb?set=set4&bgset=&size=400x400"
    enddate="Thu Dec 16 2021"
    menuItems="Option A,Option B,Option c"
    startdate="Sat Jan 01 2022"
  >
    <div
      class="ant-card-cover"
    >
      <div>
        <div
          class="sc-bdvvtL dIxYVf"
        >
          <div
            class="sc-gsDKAQ cOcCgz count-down-text"
            role="count-down-text"
          >
             days,  hours left ðŸ”¥
          </div>
          <div
            class="ant-btn-group ant-dropdown-button"
          >
            <button
              class="ant-btn ant-btn-default"
              type="button"
            />
            <button
              class="ant-btn ant-btn-default ant-btn-icon-only ant-dropdown-trigger"
              type="button"
            >
              <span
                aria-label="ellipsis"
                class="anticon anticon-ellipsis"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="ellipsis"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    />
  </div>
</div>
`;

exports[`Container should match with snapshot: MarketplaceCardContainer With Cover 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered sc-eCImPb eXMxhc"
    countDown="01:52:09 left ðŸ”¥"
    coverImg="https://robohash.org/a8cd9c187bb33c8a8f6e2774d71308fb?set=set4&bgset=&size=400x400"
    enddate="Thu Dec 16 2021"
    menuItems="Option A,Option B,Option c"
    startdate="Sat Jan 01 2022"
  >
    <div
      class="ant-card-cover"
    >
      <div>
        Hello
      </div>
    </div>
    <div
      class="ant-card-body"
    />
  </div>
</div>
`;

exports[`Container should match with snapshot: MarketplaceCardContainer Without menuItems 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered sc-eCImPb eXMxhc"
    countDown="01:52:09 left ðŸ”¥"
    coverImg="https://robohash.org/a8cd9c187bb33c8a8f6e2774d71308fb?set=set4&bgset=&size=400x400"
    enddate="Thu Dec 16 2021"
    startdate="Sat Jan 01 2022"
  >
    <div
      class="ant-card-cover"
    >
      <div>
        Hello
      </div>
    </div>
    <div
      class="ant-card-body"
    />
  </div>
</div>
`;

exports[`Container should match with snapshot: SpacedFlexContainer 1`] = `
<div>
  <div
    class="sc-bdvvtL dIxYVf"
  />
</div>
`;
